GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java

```diff
@@ -262,21 +257,49 @@ public void onDrawFrame(float frameTime) {
     /**
      * Gets the {@linkplain GVRScript a script} linked to the activity.
      * @return the {@link GVRScript}.
+     * @deprecated
      */
     public GVRScript getScript() {
         return mGVRScript;
     }
 
     /**
+     * Links {@linkplain GVRMain} to the activity; sets the version;
+     * 
+     * @param gvrMain
+     *            An instance of {@link GVRMain} to handle callbacks on the framework
+     *            thread.
+     * @param dataFileName
+     *            Name of the XML file containing the framebuffer parameters. 
+     *            <p>
+     *            The XML filename is relative to the application's
+     *            {@code assets} directory, and can specify a file in a
+     *            directory under the application's {@code assets} directory.
+     */
+    public void setMain(GVRMain gvrMain, String dataFileName) {
+        this.mGVRScript = gvrMain;
+        this.mGVRMain = gvrMain;
+        setScript(gvrMain, dataFileName);
```

Comment:
Ok. I understand the purpose of this patch now. It's adding GVRMain as a subclass of GVRScript and use setMain and getMain to replace setScript and getScript.